plugin/trino-hive/src/main/java/io/trino/plugin/hive/HiveMetadata.java

```diff
@@ -1474,7 +1474,7 @@ public void dropTable(ConnectorSession session, ConnectorTableHandle tableHandle
         if (metastore.getTable(handle.getSchemaName(), handle.getTableName()).isEmpty()) {
             throw new TableNotFoundException(handle.getSchemaTableName());
         }
-        metastore.dropTable(session, handle.getSchemaName(), handle.getTableName());
+        metastore.dropTable(session, handle.getSchemaName(), handle.getTableName(), true);
```

Comment:
Pls add a test for counting the metastore access operations

https://github.com/trinodb/trino/blob/fd14f9f477c48bb3d14dc3480715c792c2246617/plugin/trino-iceberg/src/test/java/io/trino/plugin/iceberg/TestIcebergMetastoreAccessOperations.java#L310-L320